src/shared_state.rs

```diff
@@ -239,6 +239,7 @@ pub(crate) struct SharedState {
     pub(crate) subs: RwLock<HashMap<usize, SubscriptionState>>,
     pub(crate) pongs: Mutex<VecDeque<Sender<bool>>>,
     pub(crate) outbound: Outbound,
+    pub(crate) info: RwLock<ServerInfo>,
```

Comment:
Do we have the ability to get CID from the info as well? Or is IP first server info that client can get?

Reply:
IP is for whatever server the client last connected to at the time that client_ip is able to lock this mutex, and the ServerInfo is updated any time a later INFO message is received.

Reply:
I meant is this the first access to the server info from the client? We probably want to surface CID as well as IP.

Reply:
https://github.com/nats-io/nats.rs/pull/30